lib/archethic/contracts.ex

```diff
@@ -255,32 +239,53 @@ defmodule Archethic.Contracts do
         ) :: {:ok, logs :: list(String.t())} | {:error, ConditionRejected.t() | Failure.t()}
   def execute_condition(
         condition_key,
-        contract = %Contract{version: version, conditions: conditions},
+        contract = %Contract{conditions: conditions},
         transaction = %Transaction{},
         maybe_recipient,
         datetime
       ) do
-    case Map.get(conditions, condition_key) do
-      nil ->
-        # only inherit condition are optional
-        if condition_key == :inherit do
-          {:ok, []}
-        else
-          {:error,
-           %Failure{
-             error: "Missing condition",
-             user_friendly_error: "Missing condition",
-             logs: [],
-             stacktrace: []
-           }}
-        end
+    conditions
+    |> Map.get(condition_key)
+    |> do_execute_condition(condition_key, contract, transaction, datetime, maybe_recipient)
+  rescue
```

Comment:
I think this rescue is not usefull anymore since the error will be rescued in the async task

Reply:
I was not sure about this also, I kept in case of.
I don't know before if the rescue was for function like ` Interpreter.get_named_action_constants`or `get_condition_constants`